### ISSUE TITLE
Training on customer data which has very short distance

### ISSUE THREAD

**Issue by SallyHuangImperial** [nianticlabs/monodepth2#235]
Hi, Thank you for sharing these excellent works! 

I have read the previous response from issues 207. You mentioned that the 'depth' is not in meters, it is only up to scale, but is generally the value you want to use. However, you defined the max distance and min distance during training. In my custom dataset, I have stereo image pairs and the distance from my object to the camera is 5~25 cm. Should I change the max and min to 0.05 and 0.25? Actually, when I change these values, the training results are very bad.  So if the 'depth' is not in meters and is just up to scale, does it mean that I can give any max and min values for training as long as the results are good? I used the stereo model. 

Thank you!

`disp` is the "raw" output of the network and is coming from a sigmoid layer so is in [0, 1]
`scaled_disp` is `disp` after being scaled and offset such that its inverse `depth` is bound between 0.1 and 100 units.
https://github.com/nianticlabs/monodepth2/blob/13200ab2f29f2f10dec3aa5db29c32a23e29d376/layers.py#L16-L25
`depth` isn't in meters, it is only up to scale, but is generally the value you want to use.
See [this previous issue](https://github.com/nianticlabs/monodepth2/issues/159#issuecomment-599712590) for more information.

I hope this helps!

_Originally posted by @mrharicot in https://github.com/nianticlabs/monodepth2/issues/207#issuecomment-653987677_

**Comment by mrharicot** [comment]
Hi,
The min and max distances were chosen so that training was stable for KITTI.
They're indeed not meters, however we set the baseline to be 0.1 units, so if you change the min and max depth, you should also change the baseline to be consistent:
https://github.com/nianticlabs/monodepth2/blob/101dd03f00e0d8e9bcd8a8d760f94bc2820f5549/datasets/mono_dataset.py#L196
I hope this helps!

**Comment by SallyHuangImperial** [comment]
Hi,
Many thanks.
In the following code line, I saw that you just timed the baseline and didn't time the focal length (in pixels). I am confused because depth = focal_length*baseline/disparity. 
https://github.com/nianticlabs/monodepth2/blob/101dd03f00e0d8e9bcd8a8d760f94bc2820f5549/evaluate_depth.py#L156
Besides, I don't have ground truth for the depth but I have basic calibration matrix of the stereo camera. And also as the min and max distances are not meters, so how can I keep consistent between the min_max distances and the baseline?
Thank you!

**Comment by mrharicot** [comment]
If you know the baseline between your two cameras, and that you trained with the correct normalized intrinsics `self.K` in the dataloader then you can convert to metric depth by scaling the depth by the ratio of the baselines:
```python
STEREO_SCALE_FACTOR = actual_stereo_baseline_in_meters / 0.1
```

**Comment by mrharicot** [comment]
Thanks :)

**Comment by SallyHuangImperial** [comment]
> Hi, Thank you for sharing these excellent works!
> 
> I have read the previous response from issues 207. You mentioned that the 'depth' is not in meters, it is only up to scale, but is generally the value you want to use. However, you defined the max distance and min distance during training. In my custom dataset, I have stereo image pairs and the distance from my object to the camera is 5~25 cm. Should I change the max and min to 0.05 and 0.25? Actually, when I change these values, the training results are very bad. So if the 'depth' is not in meters and is just up to scale, does it mean that I can give any max and min values for training as long as the results are good? I used the stereo model.
> 
> Thank you!
> 
> `disp` is the "raw" output of the network and is coming from a sigmoid layer so is in [0, 1]
> `scaled_disp` is `disp` after being scaled and offset such that its inverse `depth` is bound between 0.1 and 100 units.
> https://github.com/nianticlabs/monodepth2/blob/13200ab2f29f2f10dec3aa5db29c32a23e29d376/layers.py#L16-L25
> 
> 
> `depth` isn't in meters, it is only up to scale, but is generally the value you want to use.
> See [this previous issue](https://github.com/nianticlabs/monodepth2/issues/159#issuecomment-599712590) for more information.
> I hope this helps!
> 
> _Originally posted by @mrharicot in [#207 (comment)](https://github.com/nianticlabs/monodepth2/issues/207#issuecomment-653987677)_

Hi, I am still working in this field:)

Now I have a new dataset with ground truth, for which the depth range is between 5mm and 260mm and the baseline is about 4.9mm. Actually, if I use this code and set the baseline to 0.1 and the depth range to 5-260, I cannot obtain very accurate metric depth by multiplying 49. However, when I use the median or mean factor to multiply my depth output, the results make more sense. I want to ask how did you find the relationship between the hardcode baseline and the depth range when you were working with KITTI? And how did you find 0.1 as a proper baseline value for training?
I use the stereo mode to train my custom dataset. From the previous issues I know that if training with stereo mode, the metric depth can be obtained. I failed to recover the metric depth by just multiply 49 so I want to ask, is it OK if recover the scale by calculating the mean or median factor?
Thank you!
